lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -150,6 +153,7 @@ def configure(conf)
         end
         if @vm_id.nil? && metadata.key?('instanceId')
           @vm_id = metadata['instanceId']
+        @vm_name = Socket.gethostname if @vm_name.nil?
```

Comment:
I think we should do this regardless of platform, so maybe just move it outside of the 'case' altogether (and eliminate the other instance above)?